tests/modules/social_groups/resources/test_social_group.py

```diff
@@ -122,39 +140,54 @@ def test_basic_operation(
     module_unavailable('social_groups'), reason='SocialGroup module disabled'
 )
 def test_error_config(flask_app_client, researcher_1, admin_user):
+    import uuid
+
     valid_data = [
-        {'label': 'Matriarch', 'multipleInGroup': False},
-        {'label': 'IrritatingGit', 'multipleInGroup': True},
+        {'guid': str(uuid.uuid4()), 'label': 'Matriarch', 'multipleInGroup': False},
+        {'guid': str(uuid.uuid4()), 'label': 'IrritatingGit', 'multipleInGroup': True},
     ]
 
     # Valid data but non admin user so should fail
     error = "You don't have the permission to access the requested resource."
     soc_group_utils.set_roles(flask_app_client, researcher_1, valid_data, 403, error)
 
     missing_field = [
-        {'label': 'Matriarch'},
-        {'label': 'IrritatingGit', 'multipleInGroup': True},
+        {'guid': str(uuid.uuid4()), 'label': 'Matriarch'},
+        {'guid': str(uuid.uuid4()), 'label': 'IrritatingGit', 'multipleInGroup': True},
     ]
 
     resp = soc_group_utils.set_roles(flask_app_client, admin_user, missing_field, 400)
     # cant rely on order of keys so just test something
     assert 'Role dictionary must have the following keys' in resp.json['message']
 
     extra_field = [
-        {'label': 'Matriarch', 'multipleInGroup': False, 'attitude': True},
-        {'label': 'IrritatingGit', 'multipleInGroup': True},
+        {
+            'guid': str(uuid.uuid4()),
+            'label': 'Matriarch',
+            'multipleInGroup': False,
+            'attitude': True,
+        },
+        {'guid': str(uuid.uuid4()), 'label': 'IrritatingGit', 'multipleInGroup': True},
     ]
 
     resp = soc_group_utils.set_roles(flask_app_client, admin_user, extra_field, 400)
     assert 'Role dictionary must have the following keys' in resp.json['message']
 
     duplicate_label = [
-        {'label': 'Matriarch', 'multipleInGroup': False},
-        {'label': 'Matriarch', 'multipleInGroup': True},
+        {'guid': str(uuid.uuid4()), 'label': 'Matriarch', 'multipleInGroup': False},
+        {'guid': str(uuid.uuid4()), 'label': 'Matriarch', 'multipleInGroup': True},
     ]
     error = 'can only have Matriarch once'
     soc_group_utils.set_roles(flask_app_client, admin_user, duplicate_label, 400, error)
 
+    missing_guid = [
+        {'guid': str(uuid.uuid4()), 'label': 'Matriarch'},
+        {'label': 'Mild mannered janitor'},
+    ]
+    resp = soc_group_utils.set_roles(flask_app_client, admin_user, missing_guid, 400)
+    # cant rely on order of keys so just test something
+    assert 'Role dictionary must have the following keys' in resp.json['message']
```

Comment:
It seems like relying on non-language-agnostic error verbiage might be brittle in the long term?